crates/burn-import/src/burn/node/binary.rs

```diff
@@ -131,6 +131,7 @@ impl BinaryNode {
             (Type::Tensor(_), Type::Tensor(_)) => move |lhs, rhs| quote! { #lhs.sub(#rhs) },
             (Type::Tensor(_), Type::Scalar(_)) => move |lhs, rhs| quote! { #lhs.sub_scalar(#rhs) },
             (Type::Scalar(_), Type::Scalar(_)) => move |lhs, rhs| quote! { #lhs - #rhs },
+            (Type::Scalar(_), Type::Tensor(_)) => move |lhs, rhs| quote! { -#rhs.sub_scalar(#lhs) },
```

Comment:
Thank you! 

FYI @nathanielsimard , @laggui , @louisfd

Another reason to support native Scalar type in Burn. See our earlier design discussion: https://github.com/tracel-ai/burn/issues/1689#issuecomment-2078212769

`{ -#rhs.sub_scalar(#lhs) },` solution to `scalar - tensor` will result in two operations instead of one.